src/general/OptionsMenu.cs

```diff
@@ -461,6 +468,25 @@ private void CompareSettings()
         }
     }
 
+    private void LoadLanguages(OptionButton optionButton)
+    {
+        Array locals = TranslationServer.GetLoadedLocales();
+
+        foreach (string local in locals)
+        {
+            optionButton.AddItem(local);
+        }
+
+        if (optionButton.Items.Count == 0)
+            return;
+
+        // The option button seems to have 5 items / options.
+        // This means that the second option has the index 5 in the items list, but Selected need the id 1 to works.
+        // This is strange.
+        optionButton.Selected = optionButton.Items.IndexOf(Settings.Instance.SelectedLanguage) / 5;
+        TranslationServer.SetLocale(Settings.Instance.SelectedLanguage);
```

Comment:
This should be in a new apply method in Settings.
Also it would be very nice to also set C#'s CurrentLocale as that affects decimal separator etc. so that would be consistent with the selected language in the game.